### PR TITLE
Allow ACL to be owned by GrantedAuthoritySid

### PR DESCRIPTION
**Pull request by bberto**
Allow ACL to be owned by GrantedAuthoritySid

Closes gh-9425

### CODE REVIEW

**Comment by jgrandja** [comment]
Closing as per [comment](https://github.com/spring-projects/spring-security/issues/9425#issuecomment-784264364)

File: acl/src/test/java/org/springframework/security/acls/domain/AclAuthorizationStrategyImplTests.java [hunk 1]
```diff
@@ -41,15 +43,12 @@
 	@Mock
 	Acl acl;
 
-	GrantedAuthority authority;
+	GrantedAuthority authority = new SimpleGrantedAuthority("ROLE_SYSTEM_ADMIN");
 
 	AclAuthorizationStrategyImpl strategy;
 
-	@Before
-	public void setup() {
```
Comment (jgrandja) [inline acl/src/test/java/org/springframework/security/acls/domain/AclAuthorizationStrategyImplTests.java:49]: Please revert this change and preserve the `@Before` method and create a `withUserRoles()` static helper method  instead.

**Review by jgrandja — CHANGES_REQUESTED** [review]
Thanks for the PR @bberto. Please see review comment.

File: acl/src/test/java/org/springframework/security/acls/domain/AclAuthorizationStrategyImplTests.java (hunk 1 above)
Comment (bberto) [reply acl/src/test/java/org/springframework/security/acls/domain/AclAuthorizationStrategyImplTests.java]: switching back to `@Before` method I realized that helper method is not needed

**Comment by jgrandja** [comment]
Thanks for the PR @bberto. This is now in master.
